execution_engine/src/system/handle_payment/internal.rs

```diff
@@ -106,81 +107,89 @@ pub(crate) fn finalize_payment<P: MintProvider + RuntimeProvider + StorageProvid
         return Err(Error::InsufficientPaymentForAmountSpent);
     }
 
-    let (refund_amount, validator_reward) =
-        calculate_refund_and_reward(gas_spent, payment_amount, provider.refund_handling())?;
+    let (refund_amount, validator_fee) =
```

Comment:
I think `(refund, fee)` are better names.  I mainly dislike `validator_fee` as that implies it's a fee being charged to the validator, rather than a fee being paid to the validator.